### PR TITLE
Feature vs3 esp32 i2 c class

### PR DESCRIPTION
**Pull request by RoxanneMango**
- Added I2cEsp class with dummy functions for write() and print()
- Already populated constructor and destructor
- Added member variables such as tx/rx buffer
- Implemented read() function

### CODE REVIEW

File: ESP/I2C_ESP32/main/I2C_ESP.hpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+class I2cEsp {
+private:
+	int dataBufferLength = 512; 				///< Data buffer length
+	int txBufferLength = 2 * dataBufferLength;	///< Transmission buffer length
+	int rxBufferLength = 2 * dataBufferLength;	///< Receiving buffer length
+	
+	uint8_t * dataBuffer;	///< Data buffer
+	uint8_t * txBuffer;		///< Tranmission buffer
+	uint8_t * rxBuffer;		///< Receive buffer
```
Comment (NielsdeWaal) [inline ESP/I2C_ESP32/main/I2C_ESP.hpp:29]: Raw pointers are reasonable dangerous, try using STL pointers.

File: ESP/I2C_ESP32/main/I2C_ESP.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	
+	bool isMaster;			///< Operating mode: true = master; false = slave
+	bool checkAck;			///< Whether to check for ACKs
+	
+	int masterClockFrequency = 100000;
+	uint8_t slaveAddress = 0x28;
```
Comment (NielsdeWaal) [inline ESP/I2C_ESP32/main/I2C_ESP.hpp]: Both can be constexpr.

File: ESP/I2C_ESP32/main/main.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#include "LED.hpp"
+#include "I2C_ESP.hpp"
+
+extern "C" {
+   void app_main();
+}
+
+void app_main() {
+    auto led = LED();
+    led.blinkInfinitely(1000);
```
Comment (NielsdeWaal) [inline ESP/I2C_ESP32/main/main.cpp]: Program will stop at this point, it is an infinite loop.

File: .clang-format [hunk 4]
```diff
@@ -1,6 +1,4 @@
 ---
-AlignConsecutiveAssignments: true
```
Comment (NielsdeWaal) [inline .clang-format:2]: This line should be added back into the file.

File: ESP/I2C_ESP32/main/I2C_ESP.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	} else {
+		config.mode = I2C_MODE_SLAVE;
+		config.slave.addr_10bit_en = 0;
+		config.slave.slave_addr = slaveAddress;
+	}
+	
+	i2c_param_config(portNum, &config);
+	i2c_driver_install(portNum, config.mode, rxBufferLength, txBufferLength, 0);
+	
+	dataBuffer = new uint8_t[dataBufferLength];
```
Comment (NielsdeWaal) [inline ESP/I2C_ESP32/main/I2C_ESP.cpp:24]: Maybe don't use ```new``` in the constructor. Try moving it to class header, also see if std::array can be used, for more safe programming.

**Review by NielsdeWaal — CHANGES_REQUESTED** [review]
Try looking at these points.

File: ESP/I2C_ESP32/main/I2C_ESP.hpp (hunk 2 above)
Comment (RoxanneMango) [reply ESP/I2C_ESP32/main/I2C_ESP.hpp]: Turned them into static constexpr :+1:

File: ESP/I2C_ESP32/main/I2C_ESP.hpp (hunk 1 above)
Comment (RoxanneMango) [reply ESP/I2C_ESP32/main/I2C_ESP.hpp:29]: I2C ESP32 IDF library functions only accept raw pointers

File: .clang-format (hunk 4 above)
Comment (RoxanneMango) [reply .clang-format:2]: Added :ok_hand:

File: ESP/I2C_ESP32/main/I2C_ESP.cpp (hunk 5 above)
Comment (RoxanneMango) [reply ESP/I2C_ESP32/main/I2C_ESP.cpp:24]: Array has been tried, but would cause problems with buffers

File: ESP/I2C_ESP32/main/main.cpp (hunk 3 above)
Comment (RoxanneMango) [reply ESP/I2C_ESP32/main/main.cpp]: LED code has been commented out for the time being.

**Review by NielsdeWaal — APPROVED** [review]

**Pull request merged**